sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/utils.scala

```diff
@@ -86,20 +86,40 @@ object Utils {
       aggregateExpressions: Seq[AggregateExpression],
       aggregateFunctionToAttribute: Map[(AggregateFunction, Boolean), Attribute],
       resultExpressions: Seq[NamedExpression],
+      skipUnnecessaryAggregate: Boolean,
```

Comment:
I think it would be clearer if you called the `partialAggregation`.  Its not unnecessary, its an optimization in most cases.